[Automation] javac compile fails for org.sonatype.sisu:sisu-inject-bean:2.4.1

## javac compile on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/28121378337
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.sonatype.sisu:sisu-inject-bean:2.4.0
Raw versions JSON: [
  "2.4.1",
  "2.4.2",
  "2.4.3",
  "2.5.0",
  "2.5.1",
  "2.5.2",
  "2.5.3",
  "2.6.0"
]
Parsed version count: 8
Parsed versions: 2.4.1 2.4.2 2.4.3 2.5.0 2.5.1 2.5.2 2.5.3 2.6.0
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.sonatype.sisu:sisu-inject-bean:2.4.0:2.4.1
========================================================================================
========================================================================================
 org.sonatype.sisu:sisu-inject-bean:2.4.0:2.4.1 stage native-image run
========================================================================================
Starting stage 'native-image run' for 2.4.1 at 2026-06-24T18:49:22Z
Command: env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.sonatype.sisu:sisu-inject-bean:2.4.0
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java:44: error: method hear in class QualifiedTypeBinder cannot be applied to given types;

            new QualifiedTypeBinder(binder).hear(null, InstalledPrivateConstructorModule.class, SOURCE);
[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/build/reports/problems/problems-report.html
                                           ^

  required: Class,Object
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.
  found:    <null>,Class<InstalledPrivateConstructorModule>,Object

  reason: actual and formal argument lists differ in length
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/URLClassSpaceTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/URLClassSpaceTest.java uses or overrides a deprecated API.
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java:44: error: method hear in class QualifiedTypeBinder cannot be applied to given types;
14 actionable tasks: 6 executed, 8 up-to-date
              new QualifiedTypeBinder(binder).hear(null, InstalledPrivateConstructorModule.class, SOURCE);
                                             ^
    required: Class,Object
    found:    <null>,Class<InstalledPrivateConstructorModule>,Object
    reason: actual and formal argument lists differ in length
  Note: Recompile with -Xlint:deprecation for details.
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 5s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.sonatype.sisu:sisu-inject-bean:2.4.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 14s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.4.1 at 2026-06-24T18:49:36Z with exit code 1 after 14s
❌ ERROR: 'native-image run' for 2.4.1 failed with exit code 1.
SOFT-FAIL[native-image run][2.4.1][env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0]
========================================================================================
 org.sonatype.sisu:sisu-inject-bean:2.4.0:2.4.1 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.4.1 at 2026-06-24T18:49:36Z
Command: env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 936ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java:44: error: method hear in class QualifiedTypeBinder cannot be applied to given types;
            new QualifiedTypeBinder(binder).hear(null, InstalledPrivateConstructorModule.class, SOURCE);
                                           ^
  required: Class,Object
  found:    <null>,Class<InstalledPrivateConstructorModule>,Object
  reason: actual and formal argument lists differ in length
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/URLClassSpaceTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error

FAILURE: Build failed with an exception.

* What went wrong:

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/build/reports/problems/problems-report.html
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/URLClassSpaceTest.java uses or overrides a deprecated API.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java:44: error: method hear in class QualifiedTypeBinder cannot be applied to given types;
              new QualifiedTypeBinder(binder).hear(null, InstalledPrivateConstructorModule.class, SOURCE);
                                             ^
    required: Class,Object
    found:    <null>,Class<InstalledPrivateConstructorModule>,Object
    reason: actual and formal argument lists differ in length
  Note: Recompile with -Xlint:deprecation for details.
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.4.1 at 2026-06-24T18:49:40Z with exit code 1 after 4s
❌ ERROR: 'javac compile' for 2.4.1 failed with exit code 1.
FAILED[javac compile][2.4.1][env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/28121378337
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.sonatype.sisu:sisu-inject-bean:2.4.0
Raw versions JSON: [
  "2.4.1",
  "2.4.2",
  "2.4.3",
  "2.5.0",
  "2.5.1",
  "2.5.2",
  "2.5.3",
  "2.6.0"
]
Parsed version count: 8
Parsed versions: 2.4.1 2.4.2 2.4.3 2.5.0 2.5.1 2.5.2 2.5.3 2.6.0
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.sonatype.sisu:sisu-inject-bean:2.4.0:2.4.1
========================================================================================
========================================================================================
 org.sonatype.sisu:sisu-inject-bean:2.4.0:2.4.1 stage native-image run
========================================================================================
Starting stage 'native-image run' for 2.4.1 at 2026-06-24T18:49:19Z
Command: env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.sonatype.sisu:sisu-inject-bean:2.4.0
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java:44: error: method hear in class QualifiedTypeBinder cannot be applied to given types;
            new QualifiedTypeBinder(binder).hear(null, InstalledPrivateConstructorModule.class, SOURCE);
                                           ^
  required: Class,Object
  found:    <null>,Class<InstalledPrivateConstructorModule>,Object
  reason: actual and formal argument lists differ in length
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/URLClassSpaceTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/URLClassSpaceTest.java uses or overrides a deprecated API.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java:44: error: method hear in class QualifiedTypeBinder cannot be applied to given types;
              new QualifiedTypeBinder(binder).hear(null, InstalledPrivateConstructorModule.class, SOURCE);
                                             ^
    required: Class,Object
    found:    <null>,Class<InstalledPrivateConstructorModule>,Object
    reason: actual and formal argument lists differ in length
  Note: Recompile with -Xlint:deprecation for details.
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
14 actionable tasks: 6 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.sonatype.sisu:sisu-inject-bean:2.4.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 13s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.4.1 at 2026-06-24T18:49:33Z with exit code 1 after 14s
❌ ERROR: 'native-image run' for 2.4.1 failed with exit code 1.
SOFT-FAIL[native-image run][2.4.1][env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean test -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0]
========================================================================================
 org.sonatype.sisu:sisu-inject-bean:2.4.0:2.4.1 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.4.1 at 2026-06-24T18:49:33Z
Command: env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 891ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java:44: error: method hear in class QualifiedTypeBinder cannot be applied to given types;
            new QualifiedTypeBinder(binder).hear(null, InstalledPrivateConstructorModule.class, SOURCE);
                                           ^
  required: Class,Object
  found:    <null>,Class<InstalledPrivateConstructorModule>,Object
  reason: actual and formal argument lists differ in length
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/URLClassSpaceTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
10 actionable tasks: 1 executed, 9 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/URLClassSpaceTest.java uses or overrides a deprecated API.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java:44: error: method hear in class QualifiedTypeBinder cannot be applied to given types;
              new QualifiedTypeBinder(binder).hear(null, InstalledPrivateConstructorModule.class, SOURCE);
                                             ^
    required: Class,Object
    found:    <null>,Class<InstalledPrivateConstructorModule>,Object
    reason: actual and formal argument lists differ in length
  Note: Recompile with -Xlint:deprecation for details.
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.4.1 at 2026-06-24T18:49:36Z with exit code 1 after 3s
❌ ERROR: 'javac compile' for 2.4.1 failed with exit code 1.
FAILED[javac compile][2.4.1][env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean compileTestJava -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0]
```

## javac compile on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: javac compile
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/28121378337
Last 300 lines of the log:
```console
========================================================================================
 run-consecutive-tests input
========================================================================================
Coordinates: org.sonatype.sisu:sisu-inject-bean:2.4.0
Raw versions JSON: [
  "2.4.1",
  "2.4.2",
  "2.4.3",
  "2.5.0",
  "2.5.1",
  "2.5.2",
  "2.5.3",
  "2.6.0"
]
Parsed version count: 8
Parsed versions: 2.4.1 2.4.2 2.4.3 2.5.0 2.5.1 2.5.2 2.5.3 2.6.0
Timeout per Gradle invocation: 5m
========================================================================================
 Testing org.sonatype.sisu:sisu-inject-bean:2.4.0:2.4.1
========================================================================================
========================================================================================
 org.sonatype.sisu:sisu-inject-bean:2.4.0:2.4.1 stage native-image run
========================================================================================
Starting stage 'native-image run' for 2.4.1 at 2026-06-24T18:49:25Z
Command: env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
Starting a Gradle Daemon, 1 busy Daemon could not be reused, use --status for details
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean UP-TO-DATE

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: org.sonatype.sisu:sisu-inject-bean:2.4.0
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar

> Task :compileTestJava FAILED
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java:44: error: method hear in class QualifiedTypeBinder cannot be applied to given types;
            new QualifiedTypeBinder(binder).hear(null, InstalledPrivateConstructorModule.class, SOURCE);
                                           ^
  required: Class,Object
  found:    <null>,Class<InstalledPrivateConstructorModule>,Object
  reason: actual and formal argument lists differ in length
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/URLClassSpaceTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/URLClassSpaceTest.java uses or overrides a deprecated API.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java:44: error: method hear in class QualifiedTypeBinder cannot be applied to given types;
              new QualifiedTypeBinder(binder).hear(null, InstalledPrivateConstructorModule.class, SOURCE);
                                             ^
    required: Class,Object
    found:    <null>,Class<InstalledPrivateConstructorModule>,Object
    reason: actual and formal argument lists differ in length

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 6 executed, 8 up-to-date
  Note: Recompile with -Xlint:deprecation for details.
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 6s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.sonatype.sisu:sisu-inject-bean:2.4.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 15s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 2.4.1 at 2026-06-24T18:49:41Z with exit code 1 after 16s
❌ ERROR: 'native-image run' for 2.4.1 failed with exit code 1.
SOFT-FAIL[native-image run][2.4.1][env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0]
========================================================================================
 org.sonatype.sisu:sisu-inject-bean:2.4.0:2.4.1 stage javac compile
========================================================================================
Starting stage 'javac compile' for 2.4.1 at 2026-06-24T18:49:41Z
Command: env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 964ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE

> Task :compileTestJava FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/build/reports/problems/problems-report.html
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java:44: error: method hear in class QualifiedTypeBinder cannot be applied to given types;
            new QualifiedTypeBinder(binder).hear(null, InstalledPrivateConstructorModule.class, SOURCE);
                                           ^
  required: Class,Object
  found:    <null>,Class<InstalledPrivateConstructorModule>,Object
  reason: actual and formal argument lists differ in length
Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/URLClassSpaceTest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed; see the compiler output below.
  Note: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/URLClassSpaceTest.java uses or overrides a deprecated API.
  /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.0/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java:44: error: method hear in class QualifiedTypeBinder cannot be applied to given types;
              new QualifiedTypeBinder(binder).hear(null, InstalledPrivateConstructorModule.class, SOURCE);
                                             ^
    required: Class,Object
    found:    <null>,Class<InstalledPrivateConstructorModule>,Object

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
    reason: actual and formal argument lists differ in length

  Note: Recompile with -Xlint:deprecation for details.
  1 error

* Try:
> Check your code and dependencies to fix the compilation error(s)
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.> Run with --scan to generate a Build Scan (Powered by Develocity).

10 actionable tasks: 1 executed, 9 up-to-date

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Compilation failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 2.4.1 at 2026-06-24T18:49:45Z with exit code 1 after 4s
❌ ERROR: 'javac compile' for 2.4.1 failed with exit code 1.
FAILED[javac compile][2.4.1][env GVM_TCK_LV=2.4.1 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean compileTestJava -Pcoordinates=org.sonatype.sisu:sisu-inject-bean:2.4.0]
```

--- a/metadata/org.sonatype.sisu/sisu-inject-bean/2.4.1/reachability-metadata.json
+++ b/metadata/org.sonatype.sisu/sisu-inject-bean/2.4.1/reachability-metadata.json
@@ -1,0 +1,19 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "org.eclipse.sisu.space.QualifiedTypeBinder"
+      },
+      "type": "com.google.inject.util.Modules$1"
+    },
+    {
+      "type": "com.google.inject.internal.InjectorShell$RootModule"
+    },
+    {
+      "condition": {
+        "typeReached": "org.eclipse.sisu.space.URLClassSpace"
+      },
+      "type": "org.sonatype.guice.bean.reflect.URLClassSpace"
+    }
+  ]
+}

--- a/metadata/org.sonatype.sisu/sisu-inject-bean/2.4.1/reachability-metadata.json
+++ b/metadata/org.sonatype.sisu/sisu-inject-bean/2.4.1/reachability-metadata.json
@@ -1,19 +1,19 @@
 {
-  "reflection": [
+  "reflection" : [
     {
-      "condition": {
-        "typeReached": "org.eclipse.sisu.space.QualifiedTypeBinder"
+      "condition" : {
+        "typeReached" : "org.eclipse.sisu.space.QualifiedTypeBinder"
       },
-      "type": "com.google.inject.util.Modules$1"
+      "type" : "com.google.inject.util.Modules$1"
     },
     {
-      "type": "com.google.inject.internal.InjectorShell$RootModule"
+      "type" : "com.google.inject.internal.InjectorShell$RootModule"
     },
     {
-      "condition": {
-        "typeReached": "org.eclipse.sisu.space.URLClassSpace"
+      "condition" : {
+        "typeReached" : "org.eclipse.sisu.space.URLClassSpace"
       },
-      "type": "org.sonatype.guice.bean.reflect.URLClassSpace"
+      "type" : "org.sonatype.guice.bean.reflect.URLClassSpace"
     }
   ]
 }

--- a/metadata/org.sonatype.sisu/sisu-inject-bean/index.json
+++ b/metadata/org.sonatype.sisu/sisu-inject-bean/index.json
@@ -51,5 +51,17 @@
       "org.aopalliance",
       "org.sonatype"
     ]
+  },
+  {
+    "metadata-version" : "2.4.1",
+    "tested-versions" : [
+      "2.4.1"
+    ],
+    "allowed-packages" : [
+      "javax.enterprise.inject",
+      "javax.inject",
+      "org.aopalliance",
+      "org.sonatype"
+    ]
   }
 ]

--- a/metadata/org.sonatype.sisu/sisu-inject-bean/index.json
+++ b/metadata/org.sonatype.sisu/sisu-inject-bean/index.json
@@ -54,6 +54,11 @@
   },
   {
     "metadata-version" : "2.4.1",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/sonatype/sisu/sisu-inject-bean/$version$/sisu-inject-bean-$version$-sources.jar",
+    "repository-url" : "https://github.com/sonatype/sisu",
+    "test-code-url" : "N/A",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/org/sonatype/sisu/sisu-inject-bean/$version$/sisu-inject-bean-$version$-javadoc.jar",
+    "description" : "Sisu Inject Bean is a legacy wrapper around Sonatype Sisu's JSR-330 bean support, packaging Guice-based bean discovery, binding, locating, conversion, and container integration classes for Java applications. It helps applications scan class spaces for annotated components and wire them into Guice injectors while exposing Sonatype-specific injection annotations and utilities.",
     "tested-versions" : [
       "2.4.1"
     ],
@@ -61,6 +66,7 @@
       "javax.enterprise.inject",
       "javax.inject",
       "org.aopalliance",
+      "org.eclipse.sisu",
       "org.sonatype"
     ]
   }

--- a/stats/org.sonatype.sisu/sisu-inject-bean/2.4.1/execution-metrics.json
+++ b/stats/org.sonatype.sisu/sisu-inject-bean/2.4.1/execution-metrics.json
@@ -1,0 +1,106 @@
+{
+  "fix_javac_fail:2026-06-26": {
+    "timestamp": "2026-06-26T12:39:51.513539Z",
+    "library": "org.sonatype.sisu:sisu-inject-bean:2.4.1",
+    "starting_commit": "7d7a647370123763419ef88eb054e9ee3e632280",
+    "ending_commit": "caee3167b1edde072718c3b7e05d655bfee90c7a",
+    "previous_library": "org.sonatype.sisu:sisu-inject-bean:1.4.2",
+    "strategy_name": "javac_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "2.4.1",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 2,
+            "totalCalls": 2
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 2,
+        "totalCalls": 2
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 51,
+          "missed": 802,
+          "ratio": 0.059789,
+          "total": 853
+        },
+        "line": {
+          "covered": 10,
+          "missed": 215,
+          "ratio": 0.044444,
+          "total": 225
+        },
+        "method": {
+          "covered": 5,
+          "missed": 128,
+          "ratio": 0.037594,
+          "total": 133
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "1.4.2",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 8,
+            "totalCalls": 8
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 2,
+            "totalCalls": 2
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 10,
+        "totalCalls": 10
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 477,
+          "missed": 13196,
+          "ratio": 0.034886,
+          "total": 13673
+        },
+        "line": {
+          "covered": 111,
+          "missed": 1129,
+          "ratio": 0.089516,
+          "total": 1240
+        },
+        "method": {
+          "covered": 38,
+          "missed": 459,
+          "ratio": 0.076459,
+          "total": 497
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 150712,
+      "cached_input_tokens_used": 900608,
+      "output_tokens_used": 9594,
+      "iterations": 3,
+      "cost_usd": 0.7381,
+      "tested_library_loc": 10,
+      "metadata_entries": 3,
+      "test_only_metadata_entries": 1,
+      "previous_library_metadata_entries": 2,
+      "previous_library_test_only_metadata_entries": 1,
+      "code_coverage_percent": 4.44,
+      "previous_library_coverage_percent": 8.95
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/DeclaredMembersInnerViewAnonymous1Test.java",
+      "metadata_file": "metadata/org.sonatype.sisu/sisu-inject-bean/2.4.1/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.sonatype.sisu/sisu-inject-bean/2.4.1/stats.json
+++ b/stats/org.sonatype.sisu/sisu-inject-bean/2.4.1/stats.json
@@ -1,0 +1,37 @@
+{
+  "versions" : [ {
+    "version" : "2.4.1",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 2,
+          "totalCalls" : 2
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 2,
+      "totalCalls" : 2
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 51,
+        "missed" : 802,
+        "ratio" : 0.059789,
+        "total" : 853
+      },
+      "line" : {
+        "covered" : 10,
+        "missed" : 215,
+        "ratio" : 0.044444,
+        "total" : 225
+      },
+      "method" : {
+        "covered" : 5,
+        "missed" : 128,
+        "ratio" : 0.037594,
+        "total" : 133
+      }
+    }
+  } ]
+}

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/.gitignore
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/build.gradle
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.sonatype.sisu:sisu-inject-bean:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/gradle.properties
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.sonatype.sisu:sisu-inject-bean:2.4.1
+library.version = 2.4.1
+metadata.dir = org.sonatype.sisu/sisu-inject-bean/2.4.1/

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/settings.gradle
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.sonatype.sisu.sisu-inject-bean_tests'

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/BeanPropertyFieldTest.java
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/BeanPropertyFieldTest.java
@@ -1,0 +1,40 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_sonatype_sisu.sisu_inject_bean;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.sonatype.guice.bean.reflect.BeanProperties;
+import org.sonatype.guice.bean.reflect.BeanProperty;
+import org.sonatype.guice.bean.reflect.IgnoreSetters;
+
+public class BeanPropertyFieldTest {
+    @Test
+    void setUpdatesPrivateFieldDiscoveredAsBeanProperty() {
+        FieldBackedBean bean = new FieldBackedBean();
+        BeanProperty<Object> property = firstPropertyOf(FieldBackedBean.class);
+
+        property.set(bean, "configured");
+
+        assertThat(property.getName()).isEqualTo("value");
+        assertThat(bean.value()).isEqualTo("configured");
+    }
+
+    private static BeanProperty<Object> firstPropertyOf(Class<?> beanType) {
+        return new BeanProperties(beanType).iterator().next();
+    }
+
+    @IgnoreSetters
+    private static final class FieldBackedBean {
+        private String value;
+
+        private String value() {
+            return value;
+        }
+    }
+}

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/BeanPropertyFieldTest.java
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/BeanPropertyFieldTest.java
@@ -8,10 +8,9 @@ package org_sonatype_sisu.sisu_inject_bean;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
+import org.eclipse.sisu.bean.BeanProperties;
+import org.eclipse.sisu.bean.BeanProperty;
 import org.junit.jupiter.api.Test;
-import org.sonatype.guice.bean.reflect.BeanProperties;
-import org.sonatype.guice.bean.reflect.BeanProperty;
-import org.sonatype.guice.bean.reflect.IgnoreSetters;
 
 public class BeanPropertyFieldTest {
     @Test
@@ -29,7 +28,6 @@ public class BeanPropertyFieldTest {
         return new BeanProperties(beanType).iterator().next();
     }
 
-    @IgnoreSetters
     private static final class FieldBackedBean {
         private String value;
 

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/BeanPropertySetterTest.java
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/BeanPropertySetterTest.java
@@ -1,0 +1,55 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_sonatype_sisu.sisu_inject_bean;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.sonatype.guice.bean.reflect.BeanProperties;
+import org.sonatype.guice.bean.reflect.BeanProperty;
+
+public class BeanPropertySetterTest {
+    @Test
+    void setInvokesDiscoveredPrivateSetter() {
+        SetterBackedBean bean = new SetterBackedBean();
+        BeanProperty<Object> property = propertyNamed(SetterBackedBean.class, "configured");
+
+        property.set(bean, "configured-value");
+
+        assertThat(property.getName()).isEqualTo("configured");
+        assertThat(bean.configuredValue()).isEqualTo("configured-value");
+        assertThat(bean.setterInvocationCount()).isEqualTo(1);
+    }
+
+    private static BeanProperty<Object> propertyNamed(Class<?> beanType, String name) {
+        for (BeanProperty<Object> property : new BeanProperties(beanType)) {
+            if (name.equals(property.getName())) {
+                return property;
+            }
+        }
+        throw new AssertionError("No bean property named " + name);
+    }
+
+    private static final class SetterBackedBean {
+        private String configuredValue;
+
+        private int setterInvocationCount;
+
+        private void setConfigured(String configuredValue) {
+            this.configuredValue = configuredValue;
+            setterInvocationCount++;
+        }
+
+        private String configuredValue() {
+            return configuredValue;
+        }
+
+        private int setterInvocationCount() {
+            return setterInvocationCount;
+        }
+    }
+}

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/BeanPropertySetterTest.java
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/BeanPropertySetterTest.java
@@ -8,9 +8,9 @@ package org_sonatype_sisu.sisu_inject_bean;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
+import org.eclipse.sisu.bean.BeanProperties;
+import org.eclipse.sisu.bean.BeanProperty;
 import org.junit.jupiter.api.Test;
-import org.sonatype.guice.bean.reflect.BeanProperties;
-import org.sonatype.guice.bean.reflect.BeanProperty;
 
 public class BeanPropertySetterTest {
     @Test

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/DeclaredMembersInnerViewAnonymous1Test.java
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/DeclaredMembersInnerViewAnonymous1Test.java
@@ -13,9 +13,9 @@ import java.lang.reflect.Member;
 import java.util.ArrayList;
 import java.util.List;
 
+import org.eclipse.sisu.bean.DeclaredMembers;
+import org.eclipse.sisu.bean.DeclaredMembers.View;
 import org.junit.jupiter.api.Test;
-import org.sonatype.guice.bean.reflect.DeclaredMembers;
-import org.sonatype.guice.bean.reflect.DeclaredMembers.View;
 
 public class DeclaredMembersInnerViewAnonymous1Test {
     @Test

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/DeclaredMembersInnerViewAnonymous1Test.java
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/DeclaredMembersInnerViewAnonymous1Test.java
@@ -1,0 +1,49 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_sonatype_sisu.sisu_inject_bean;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Member;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import org.sonatype.guice.bean.reflect.DeclaredMembers;
+import org.sonatype.guice.bean.reflect.DeclaredMembers.View;
+
+public class DeclaredMembersInnerViewAnonymous1Test {
+    @Test
+    void constructorsViewIteratesDeclaredConstructors() {
+        List<Constructor<?>> constructors = new ArrayList<>();
+
+        for (Member member : new DeclaredMembers(ConstructorFixture.class, View.CONSTRUCTORS)) {
+            assertThat(member).isInstanceOf(Constructor.class);
+            constructors.add((Constructor<?>) member);
+        }
+
+        assertThat(constructors)
+            .hasSize(3)
+            .extracting(Constructor::getParameterCount)
+            .containsExactlyInAnyOrder(0, 1, 2);
+    }
+
+    static final class ConstructorFixture {
+        ConstructorFixture() {
+        }
+
+        private ConstructorFixture(String value) {
+            assertThat(value).isNotNull();
+        }
+
+        ConstructorFixture(String value, int number) {
+            assertThat(value).isNotNull();
+            assertThat(number).isNotNegative();
+        }
+    }
+}

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java
@@ -1,0 +1,67 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_sonatype_sisu.sisu_inject_bean;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.jupiter.api.Test;
+import org.sonatype.guice.bean.binders.QualifiedTypeBinder;
+
+import com.google.inject.Binder;
+import com.google.inject.Binding;
+import com.google.inject.Key;
+import com.google.inject.Module;
+import com.google.inject.spi.Element;
+import com.google.inject.spi.Elements;
+
+public class QualifiedTypeBinderTest {
+    private static final Object SOURCE = "qualified-module-source";
+
+    @Test
+    void hearInstallsConcreteModuleByInvokingItsNoArgConstructor() {
+        InstalledPrivateConstructorModule.created().set(0);
+
+        List<Element> elements = Elements.getElements(new ScanningModule());
+
+        boolean installedBinding = elements.stream()
+            .filter(Binding.class::isInstance)
+            .map(Binding.class::cast)
+            .anyMatch(binding -> Key.get(ModuleBoundService.class).equals(binding.getKey()));
+        assertThat(InstalledPrivateConstructorModule.created().get()).isEqualTo(1);
+        assertThat(installedBinding).isTrue();
+    }
+
+    private static final class ScanningModule implements Module {
+        @Override
+        public void configure(Binder binder) {
+            new QualifiedTypeBinder(binder).hear(null, InstalledPrivateConstructorModule.class, SOURCE);
+        }
+    }
+
+    private static final class InstalledPrivateConstructorModule implements Module {
+        private static final AtomicInteger CREATED = new AtomicInteger();
+
+        private InstalledPrivateConstructorModule() {
+            CREATED.incrementAndGet();
+        }
+
+        @Override
+        public void configure(Binder binder) {
+            binder.bind(ModuleBoundService.class).toInstance(new ModuleBoundService());
+        }
+
+        private static AtomicInteger created() {
+            return CREATED;
+        }
+    }
+
+    private static final class ModuleBoundService {
+    }
+}

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/QualifiedTypeBinderTest.java
@@ -11,8 +11,8 @@ import static org.assertj.core.api.Assertions.assertThat;
 import java.util.List;
 import java.util.concurrent.atomic.AtomicInteger;
 
+import org.eclipse.sisu.space.QualifiedTypeBinder;
 import org.junit.jupiter.api.Test;
-import org.sonatype.guice.bean.binders.QualifiedTypeBinder;
 
 import com.google.inject.Binder;
 import com.google.inject.Binding;
@@ -41,7 +41,7 @@ public class QualifiedTypeBinderTest {
     private static final class ScanningModule implements Module {
         @Override
         public void configure(Binder binder) {
-            new QualifiedTypeBinder(binder).hear(null, InstalledPrivateConstructorModule.class, SOURCE);
+            new QualifiedTypeBinder(binder).hear(InstalledPrivateConstructorModule.class, SOURCE);
         }
     }
 

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/SisuGuiceAnonymous2Test.java
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/SisuGuiceAnonymous2Test.java
@@ -1,0 +1,42 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_sonatype_sisu.sisu_inject_bean;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+import org.sonatype.guice.bean.containers.SisuGuice;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Binding;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+
+public class SisuGuiceAnonymous2Test {
+    @Test
+    void enhancedInjectorDelegatesRegularInjectorMethods() {
+        Injector injector = Guice.createInjector(new ServiceModule());
+        Injector enhancedInjector = SisuGuice.enhance(injector);
+
+        Map<Key<?>, Binding<?>> bindings = enhancedInjector.getBindings();
+
+        assertThat(bindings).containsKey(Key.get(Service.class));
+    }
+
+    private static final class ServiceModule extends AbstractModule {
+        @Override
+        protected void configure() {
+            bind(Service.class).toInstance(new Service());
+        }
+    }
+
+    private static final class Service {
+    }
+}

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/SisuGuiceTest.java
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/SisuGuiceTest.java
@@ -1,0 +1,41 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_sonatype_sisu.sisu_inject_bean;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.lang.reflect.Proxy;
+
+import org.junit.jupiter.api.Test;
+import org.sonatype.guice.bean.containers.SisuGuice;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+
+public class SisuGuiceTest {
+    @Test
+    void enhanceWrapsInjectorInDynamicProxy() {
+        Injector injector = Guice.createInjector(new TestModule());
+
+        Injector enhancedInjector = SisuGuice.enhance(injector);
+
+        Class<?> enhancedType = enhancedInjector.getClass();
+        assertThat(Proxy.isProxyClass(enhancedType)).isTrue();
+        assertThat(Injector.class.isAssignableFrom(enhancedType)).isTrue();
+    }
+
+    private static final class TestModule extends AbstractModule {
+        @Override
+        protected void configure() {
+            bind(Service.class).toInstance(new Service());
+        }
+    }
+
+    private static final class Service {
+    }
+}

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/URLClassSpaceTest.java
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/java/org_sonatype_sisu/sisu_inject_bean/URLClassSpaceTest.java
@@ -1,0 +1,51 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_sonatype_sisu.sisu_inject_bean;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.net.URL;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+import org.sonatype.guice.bean.reflect.URLClassSpace;
+
+public class URLClassSpaceTest {
+    private static final String RESOURCE_NAME = "org_sonatype_sisu/sisu_inject_bean/url-class-space-resource.txt";
+
+    @Test
+    void loadClassDelegatesToBackingClassLoader() {
+        URLClassSpace classSpace = new URLClassSpace(URLClassSpaceTest.class.getClassLoader());
+
+        Class<?> loadedClass = classSpace.loadClass(URLClassSpace.class.getName());
+
+        assertThat(loadedClass).isSameAs(URLClassSpace.class);
+    }
+
+    @Test
+    void getResourceDelegatesToBackingClassLoader() {
+        URLClassSpace classSpace = new URLClassSpace(URLClassSpaceTest.class.getClassLoader());
+
+        URL resource = classSpace.getResource(RESOURCE_NAME);
+
+        assertThat(resource).isNotNull();
+        assertThat(resource.toString()).endsWith(RESOURCE_NAME);
+    }
+
+    @Test
+    void getResourcesDelegatesToBackingClassLoader() {
+        URLClassSpace classSpace = new URLClassSpace(URLClassSpaceTest.class.getClassLoader());
+
+        Enumeration<URL> resourceEnumeration = classSpace.getResources(RESOURCE_NAME);
+        List<URL> resources = Collections.list(resourceEnumeration);
+
+        assertThat(resources).isNotEmpty();
+        assertThat(resources).allSatisfy(resource -> assertThat(resource.toString()).endsWith(RESOURCE_NAME));
+    }
+}

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -1,0 +1,10 @@
+{
+  "resources" : [
+    {
+      "condition" : {
+        "typeReached" : "org.sonatype.guice.bean.reflect.URLClassSpace"
+      },
+      "glob" : "org_sonatype_sisu/sisu_inject_bean/url-class-space-resource.txt"
+    }
+  ]
+}

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/resources/org_sonatype_sisu/sisu_inject_bean/url-class-space-resource.txt
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/src/test/resources/org_sonatype_sisu/sisu_inject_bean/url-class-space-resource.txt
@@ -1,0 +1,1 @@
+resource visible to URLClassSpaceTest

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/user-code-filter.json
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/user-code-filter.json
@@ -1,0 +1,22 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "javax.enterprise.inject.**"
+    },
+    {
+      "includeClasses" : "javax.inject.**"
+    },
+    {
+      "includeClasses" : "org.aopalliance.**"
+    },
+    {
+      "includeClasses" : "org.sonatype.**"
+    },
+    {
+      "includeClasses" : "org_sonatype_sisu.sisu_inject_bean.**"
+    }
+  ]
+}

--- a/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/user-code-filter.json
+++ b/tests/src/org.sonatype.sisu/sisu-inject-bean/2.4.1/user-code-filter.json
@@ -4,16 +4,10 @@
       "excludeClasses" : "**"
     },
     {
-      "includeClasses" : "javax.enterprise.inject.**"
+      "includeClasses" : "org.sonatype.guice.bean.**"
     },
     {
-      "includeClasses" : "javax.inject.**"
-    },
-    {
-      "includeClasses" : "org.aopalliance.**"
-    },
-    {
-      "includeClasses" : "org.sonatype.**"
+      "includeClasses" : "org.sonatype.inject.**"
     },
     {
       "includeClasses" : "org_sonatype_sisu.sisu_inject_bean.**"

